libvdrtools/indy-wallet/src/wallet.rs

```diff
@@ -352,7 +353,14 @@ impl Wallet {
                 None => {
                     // no item in cache, lets retrieve it and put it in cache.
                     let metrics_fut = cache_hit_metrics.inc_cache_miss(type_);
-                    let full_options = RecordOptions::id_value_tags();
+                    let full_options = RecordOptions {
```

Comment:
Honestly, I have no idea whether setting this to `true` all the time is a good idea and I think finding out is outside the scope of this PR. 

I changed this simply because there was no way to set it to `true`, which was bad and was needed for the migration, but if the original record options don't set it to `true`, then it remains `false` and everything happens as before.

Furthermore, the scenario you described will never happen, because when the item is added to the cache, it is always passed the `type_`.